tests/api/utils/test_dict_utils.py

```diff
@@ -0,0 +1,101 @@
+from snowflake.cli.api.utils.dict_utils import deep_merge_dicts, deep_traverse
+
+
+def test_merge_dicts_empty():
+    test_dict = {}
+    deep_merge_dicts(test_dict, {})
+    assert test_dict == {}
+
+
+def test_merge_dicts_recursive_map():
+    test_dict = {"a": "a1", "b": "b1", "c": {"d": "d1", "e": "e1"}}
+    deep_merge_dicts(test_dict, {"c": {"d": "d2", "a": "a2"}})
+    assert test_dict == {"a": "a1", "b": "b1", "c": {"d": "d2", "e": "e1", "a": "a2"}}
+
+
+def test_merge_dicts_recursive_scalar_replace_map():
+    test_dict = {"a": "a1", "b": "b1", "c": {"d": "d1", "e": "e1"}}
+    deep_merge_dicts(test_dict, {"c": "c2"})
+    assert test_dict == {"a": "a1", "b": "b1", "c": "c2"}
+
+
+def test_merge_dicts_recursive_two_arrays():
+    test_dict = {"a": "a1", "b": {"c": ["c1", "c2", "c3"], "d": "d1"}}
+    deep_merge_dicts(test_dict, {"b": {"c": ["c4", "c5"]}})
+    assert test_dict == {"a": "a1", "b": {"c": ["c4", "c5"], "d": "d1"}}
+
+
+def test_deep_traverse_on_map():
+    test_struct = {
+        "scalar_key": "hello",
+        "map_key": {"key1": "value1", "key2": "value2", "key3": True},
+        "array_key": ["array1", "array2", 333, {"nestedKey1": "nestedVal1"}],
+    }
+
+    visited_elements = []
+
+    def visit_action(element):
+        visited_elements.append(element)
+
+    deep_traverse(test_struct, visit_action)
+
+    assert visited_elements == [
```

Comment:
just to confirm, the visiting order is part of the contract, yes?

Reply:
it doesn't have to be in the future. I changed this check to ignore the order.